ValueTypeMismatchError when updating a JSON field

The example from the [documentation](https://github.com/prisma/prisma-client-go/blob/e0009c5f0ab2503966014b70bc91f36ef7f90e58/docs/reference/13-json.md#write-json-data) works for `CreateOne` but not `Update`. Looks like the update mutation expects a Json type but gets a String? 

```bash
❯ go run main.go
panic: pql error: Error in query graph construction: QueryParserError(QueryParserError { path: QueryPath { segments: ["Mutation", "updateOnePost", "data"] }, error_kind: InputUnionParseError { parsing_errors: [QueryParserError { path: QueryPath { segments: ["Mutation", "updateOnePost", "data", "PostUpdateInput", "info"] }, error_kind: ValueTypeMismatchError { have: Object({"set": String("{\"content\":\"New content\"}")}), want: Json } }, QueryParserError { path: QueryPath { segments: ["Mutation", "updateOnePost", "data", "PostUncheckedUpdateInput", "info"] }, error_kind: ValueTypeMismatchError { have: Object({"set": String("{\"content\":\"New content\"}")}), want: Json } }] } })

goroutine 1 [running]:
main.main()
	/Users/mehdikharraz/workspace/prisma-demo/main.go:61 +0xb47
exit status 2
```

```prisma
datasource db {
    provider = "postgresql"
    url      = "postgresql://mehdikharraz:@localhost:5432/prisma-demo"
}

generator db {
    provider = "go run github.com/prisma/prisma-client-go"
}

model Post {
    id        Int      @id @default(autoincrement())
    createdAt DateTime @default(now())
    updatedAt DateTime @updatedAt
    title     String
    info      Json
}

```

```go
package main

import (
	"context"
	"encoding/json"

	"demo/prisma/db"
)

type PostInfo struct {
	Content string `json:"content"`
}

func main() {
	// Connect to database
	client := db.NewClient()
	if err := client.Prisma.Connect(); err != nil {
		panic(err)
	}
	defer func() {
		if err := client.Prisma.Disconnect(); err != nil {
			panic(err)
		}
	}()
	ctx := context.Background()

	// Old post info
	oldPostInfo := &PostInfo{
		Content: "Old content",
	}
	oldPostInfoBytes, err := json.Marshal(oldPostInfo)
	if err != nil {
		panic(err)
	}

	// Create Post
	createdPost, err := client.Post.CreateOne(
		db.Post.Title.Set("Post title"),
		db.Post.Info.Set(oldPostInfoBytes),
	).Exec(ctx)
	if err != nil {
		panic(err)
	}

	// New post info
	newPostInfo := &PostInfo{
		Content: "New content",
	}
	newPostInfoBytes, err := json.Marshal(newPostInfo)
	if err != nil {
		panic(err)
	}

	// Update post
	_, err = client.Post.FindUnique(
		db.Post.ID.Equals(createdPost.ID),
	).Update(
		db.Post.Info.Set(newPostInfoBytes),
	).Exec(ctx)
	if err != nil {
		panic(err)
	}
}

```

--- a/generator/templates/actions/find.gotpl
+++ b/generator/templates/actions/find.gotpl
@@ -208,7 +208,8 @@
 						{{/* TODO consider upcoming non-set methods */}}
 						field := q.field()
 						{{/* if scalar, wrap in 'set' */}}
-						if field.Value != nil {
+						_, isJson := field.Value.(types.JSON)
+						if field.Value != nil && !isJson {
 							v := field.Value
 							field.Fields = []builder.Field{
 								{

--- a/generator/templates/actions/upsert.gotpl
+++ b/generator/templates/actions/upsert.gotpl
@@ -85,7 +85,8 @@
 			{{/* TODO re-use */}}
 			field := q.field()
 			{{/* if scalar, wrap in 'set' */}}
-			if field.Value != nil {
+			_, isJson := field.Value.(types.JSON)
+			if field.Value != nil && !isJson {
 				v := field.Value
 				field.Fields = []builder.Field{
 					{

--- a/test/types/json/default_test.go
+++ b/test/types/json/default_test.go
@@ -103,6 +103,50 @@ func TestJSON(t *testing.T) {
 
 			assert.Equal(t, expected, actual)
 		},
+	}, {
+		name: "json update field",
+		run: func(t *testing.T, client *PrismaClient, ctx cx) {
+			var b JSON = []byte(`"hi"`)
+			created, err := client.User.CreateOne(
+				User.JSON.Set(b),
+				User.JSONOpt.Set(b),
+				User.ID.Set("123"),
+			).Exec(ctx)
+			if err != nil {
+				t.Fatalf("fail %s", err)
+			}
+
+			expected := &UserModel{
+				InnerUser: InnerUser{
+					ID:      "123",
+					JSON:    b,
+					JSONOpt: &b,
+				},
+			}
+
+			assert.Equal(t, expected, created)
+
+			updated, err := client.User.FindUnique(
+				User.ID.Equals("123"),
+			).Update(
+				User.JSON.Set(b),
+				User.JSONOpt.Set(b),
+			).Exec(ctx)
+			if err != nil {
+				t.Fatalf("fail %s", err)
+			}
+
+			assert.Equal(t, expected, updated)
+
+			actual, err := client.User.FindUnique(
+				User.ID.Equals("123"),
+			).Exec(ctx)
+			if err != nil {
+				t.Fatalf("fail %s", err)
+			}
+
+			assert.Equal(t, expected, actual)
+		},
 	}}
 	for _, tt := range tests {
 		tt := tt

